packages/adminrouter/extra/src/README.md

```diff
@@ -264,28 +264,174 @@ it, but it does not seem to be justified. For example:
  * `lib/auth/open.lua` uses the `.init()` pattern in order to achive different
    behaviour depending on the flavour.
 
-## Service Endpoints
-
-Admin Router allows Marathon tasks to define custom service UI and HTTP endpoints, which are made available via `<dcos-cluster>/service/<service-name>`. This can be achieved by setting the following Marathon task labels:
-
+## Service Endpoint
+Admin Router offers `/service` endpoint which enables users to access easily
+some of the task launched on DC/OS. AR internally pre-fetches data from Mesos
+and root Marathon and stores it in ternal cache for later use when routing
+the incoming request.
+
+Services can be accessed both using namespace+name and by frameworkId in case of
+Mesos frameworks. For example - let's deploy a SchmetterlingDB framework into
+`/group1/group2/schmetterlingA`. It will be accessible using its id (let's assume
+that Mesos assigned it framework ID 819aed93-4143-4291-8ced-5afb5c726803-0000):
 ```
-"labels": {
-    "DCOS_SERVICE_NAME": "service-name",
-    "DCOS_SERVICE_PORT_INDEX": "0",
-    "DCOS_SERVICE_SCHEME": "http"
-  }
+/service/819aed93-4143-4291-8ced-5afb5c726803-0000/
+```
+and using its name:
+```
+/service/group1/group2/schmetterlingA/
 ```
 
-When your container/task has its own IP (typically when running in a virtual network), `http://<dcos-cluster>/service/service-name` would be forwarded to the container/task's IP using one of the ports in the port mapping or port definition when USER networking is enabled or one of the discovery ports otherwise. When your task/container is mapped to the host, it would be forwarded to the host running the task using the one of the ports allocated to the task. The chosen port is defined by the DCOS_SERVICE_PORT_INDEX label.
-
-In order for the forwarding to work reliably across task failures, we recommend co-locating the endpoints with the task. This way, if the task is restarted on a potentially other host and with different ports, Admin Router will pick up the new labels and update the routing. NOTE: Due to caching there might be an up to 30-second delay until the new routing is working.
-
-We would recommend having only a single task setting these labels for a given `service-name`.
-In the case of multiple task instances with the same `service-name` label, Admin Router will pick one of the tasks instances deterministically, but this might make debugging issues more difficult.
+Endpoint supports WebSockets, by default disables all NGINX buffering and all
+redirects from services are rewritten in order to support `/service` prefix
+clients are using.
+
+### Intended use
+It is very important to remember that `/service` endpoint was designed to work
+with root Marathon only. Tasks launched by MarathonOnMarathon instances will not
```

Comment:
I have no idea - people very often use MoM abbreviation. Let's leave it as is and see if anybody complains.